reader.go

```diff
@@ -371,47 +388,75 @@ func (r *Reader) parseAddenda() error {
 		msg := fmt.Sprint(msgFileBatchOutside)
 		return r.parseError(&FileError{FieldName: "Addenda", Msg: msg})
 	}
-	if len(r.currentBatch.GetEntries()) == 0 {
+
+	if r.currentBatch.GetHeader().StandardEntryClassCode != "ADV" {
+		if len(r.currentBatch.GetEntries()) == 0 {
+			return r.parseError(&FileError{FieldName: "Addenda", Msg: msgFileBatchOutside})
+		}
+		entryIndex := len(r.currentBatch.GetEntries()) - 1
+		entry := r.currentBatch.GetEntries()[entryIndex]
+
+		if entry.AddendaRecordIndicator == 1 {
```

Comment:
nit: it might be easier to read if we bail out earlier.

```Go
if entry.AddendaRecordIndicator != 1 {
  return r.parseError(&FileError{...})
}
```

Reply:
That's fine, it was done that way in the original library, so I just left it, but I don't mind changing for adv.  Not critical.